src/test/java/scala/compat/java8/FutureConvertersTest.java

```diff
@@ -3,321 +3,377 @@
+            } catch (Exception ex) {
+                throw new RuntimeException(ex);
+            }
+            return v.length();
+        });
+        p.success("Hello");
+        latch.countDown();
+        assertEquals((Integer) 5, second.toCompletableFuture().get());
+    }
+
+    @Test
+    public void testToJavaExceptionally() throws InterruptedException,
+            ExecutionException {
+        final Promise<String> p = promise();
+        final CompletionStage<String> cs = toJava(p.future());
+        final CountDownLatch latch = new CountDownLatch(1);
+        final CompletionStage<String> second = cs.exceptionally(e -> {
+            try {
+                assertTrue("latch must succeed", latch.await(1, SECONDS));
+            } catch (Exception ex) {
+                throw new RuntimeException(ex);
+            }
+            return e.getMessage();
+        });
+        p.failure(new RuntimeException("Hello"));
+        latch.countDown();
+        assertEquals("Hello", second.toCompletableFuture().get());
+    }
+
+    @Test
+    public void testToJavaThenComposeWithToJavaThenAccept() throws InterruptedException,
+            ExecutionException, TimeoutException {
+        final Promise<String> p1 = promise();
+        final CompletableFuture<String> future = new CompletableFuture<>();
 
-  @Test
-  public void testToJavaRunAfterEither() throws InterruptedException,
-      ExecutionException {
-    final Promise<String> p = promise();
-    final CompletionStage<String> cs = toJava(p.future());
-    final CompletionStage<String> other = new CompletableFuture<>();
-    final CountDownLatch latch = new CountDownLatch(1);
-    final CompletionStage<Void> second = cs.runAfterEither(other, () -> {
-      try {
-        assertTrue("latch must succeed", latch.await(1, SECONDS));
-      } catch (Exception e) {
-        throw new RuntimeException(e);
-      }
-    });
-    p.success("Hello");
-    latch.countDown();
-    assertNull("result must be Void", second.toCompletableFuture().get());
-  }
+        final CompletionStage<String> second =
+                CompletableFuture.supplyAsync(() -> "Hello").
+                        thenCompose(x -> toJava(p1.future()));
 
-  @Test
-  public void testToJavaThenCompose() throws InterruptedException,
-      ExecutionException {
-    final Promise<String> p = promise();
-    final CompletionStage<String> cs = toJava(p.future());
-    final CountDownLatch latch = new CountDownLatch(1);
-    final CompletionStage<String> second = cs.thenCompose(x -> {
-      try {
-        assertTrue("latch must succeed", latch.await(1, SECONDS));
-      } catch (Exception e) {
-        throw new RuntimeException(e);
-      }
-      return CompletableFuture.completedFuture(x);
-    });
-    p.success("Hello");
-    latch.countDown();
-    assertEquals("Hello", second.toCompletableFuture().get());
-  }
+        second.handle((x, t) -> {
+            if (t != null) {
+                t.printStackTrace();
```

Comment:
I copied the test verbatim from #29. I have just pushed another commit that notes its origin and simplifies it a little.